hikari/impl/bot.py

```diff
@@ -823,6 +824,56 @@ async def wait_for(
     ) -> event_dispatcher.EventT_co:
         return await self._events.wait_for(event_type, timeout=timeout, predicate=predicate)
 
+    async def update_presence(
+        self,
+        *,
+        status: undefined.UndefinedOr[presences.Status] = undefined.UNDEFINED,
+        idle_since: undefined.UndefinedNoneOr[datetime.datetime] = undefined.UNDEFINED,
+        activity: undefined.UndefinedNoneOr[presences.Activity] = undefined.UNDEFINED,
+        afk: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+    ) -> None:
+        """Update the presence on all shards.
+
+        This call will patch the presence on each shard. This means that
+        unless you explicitly specify a parameter, the previous value will be
+        retained. This means you do not have to track the global presence
+        in your code.
+
+        Parameters
+        ----------
+        idle_since : hikari.undefined.UndefinedNoneOr[datetime.datetime]
+            The datetime that the user started being idle. If undefined, this
+            will not be changed.
+        afk : hikari.undefined.UndefinedOr[builtins.bool]
+            If `builtins.True`, the user is marked as AFK. If `builtins.False`,
+            the user is marked as being active. If undefined, this will not be
+            changed.
+        activity : hikari.undefined.UndefinedNoneOr[hikari.presences.Activity]
+            The activity to appear to be playing. If undefined, this will not be
+            changed.
+        status : hikari.undefined.UndefinedOr[hikari.presences.Status]
+            The web status to show. If undefined, this will not be changed.
+
+        !!! note
+            This will only send the update payloads to shards that are alive.
+            Any shards that are not alive will cache the new presence for
+            when they do start.
+
+        !!! note
+            If you want to set presences per shard, access the shard you wish
+            to update (e.g. by using `BotApp.shards`), and call
+            `hikari.api.shard.GatewayShard.update_presence` on that shard.
+
+            This method is simply a facade to make performing this in bulk
+            simpler.
+        """
+        coros = [
+            s.update_presence(status=status, activity=activity, idle_since=idle_since, afk=afk)
+            for s in self._shards.values()
+        ]
+
+        await asyncio.gather(*coros)
```

Comment:
Use aio.all_of here or you will get incomplete task spam possibly if something fails due to a cancelled error.